string  id
float32 distance
float32 angle